src/reconnect-strategy/mqtts.reconnect-strategy.default.ts

```diff
@@ -0,0 +1,40 @@
+import { MqttsReconnectStrategy } from './mqtts.reconnect-strategy';
+import { ConnectError } from '../errors';
+import {promisify} from 'util';
+const sleep = promisify(setTimeout);
+
+export class MqttsReconnectStrategyDefault implements MqttsReconnectStrategy {
+    /**
+     * Internal attempts counter
+     */
+    #attempts = 1;
+    constructor(
+        /**
+         * Maximum attempts amount
+         */
+        private attempts: number = 60,
+        /**
+         * Interval between attempts, milliseconds
+         */
+        private interval: number = 1000,
+    ) {}
+    should(reason?: any) {
+        if (reason instanceof ConnectError) {
+            if (['NotAuthorized', 'UnacceptableProtocolVersion', 'BadUsernameOrPassword'].includes(reason.status)) {
```

Comment:
```suggestion
            if (reason.isError) {
```
This or something different. These codes can change (and will from MQTT 3 to 5) so it's best to handle them in a function if it's more complex or like in this case through a getter.

Reply:
I'm not sure I understand this suggestion. How does this getter help to avoid string comparison? Are we just moving string comparison from one place to another?

Reply:
Sorry, I thought the error was including the packet. It should if the packet is present. You're essentially checking for the [returnCode](https://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc385349256). So you could check for the `isSuccess` getter. 


> Are we just moving string comparison from one place to another?

Still if the comparison is longer a function could make it more readable. Like `if (isBadReturnCode(packet))`.

Reply:
`isSuccess` getter doesn't fit these needs. It should reconnect if `IdentifierRejected` or `ServerUnavailable`. Also these 2 return codes are bad return codes, so a function like `isBadReturnCode` doesn't fit. Do we really need to split it like this? The `MqttsReconnectStrategyDefault` class is small enough to split it. It is only 40 lines of code, 9 of which are commends, so does it really makes any sense to split 31 lines of code into smaller chunks?

Reply:
i mean `IdentifierRejected` and `ServerUnavailable` is the only 2 cases when we need to reconnect.  Other bad return codes should **not** lead to reconnecting. For example `NotAuthorized` is the bad code too but it makes no sense to reconnect.

Reply:
> Do we really need to split it like this?

No not if it's this simple. But then it'd be better to do this (to "filter out" the ones we need):
```suggestion
            if (!['IdentifierRejected', 'ServerUnavailable'].includes(reason.status)) {
```

Reply:
BTW i don't like these string comparisons as well. I totally agree it's bad. But the current `mqtts` codebase doesn't have tools to refactor it. In order to avoid those string comparisons, it is required to refactor a lot of `mqtts` code. 

I could just extract these strings into constants. But this solution is as temporary as the current implementation because when the `mqtts.client` code will be refactored, these constants will be replaced in favor of a more well-designed solution.

I am not ready to refactor the `mqtt.client.ts` class for now, it's a lot of time. But if you have suggestions on how to avoid string comparisons in a right way that will make sense - let me know, i will do it.

Reply:
> No not if it's this simple. But then it'd be better to do this (to "filter out" the ones we need):

Yeah it makes sense!